### PR TITLE
add pagination and new query for total count

### PR DESCRIPTION
**Pull request by DianaCrainic**

### CODE REVIEW

File: cryk-ui/src/pages/Cryptocurrencies.tsx [hunk 1]
```diff
@@ -14,20 +14,26 @@ const GET_PAGINATED_CRYPTOCURRENCIES_QUERY = gql`
...
+        }
     }
 `;
 
+const CRYPTOS_PER_PAGE = 10;
+const CURRENT_PAGE = 1;
+
 export default function Cryptocurrencies() {
     const title = "Cryptocurrencies";
+    const [cryptosPerPage, setCryptosPerPage] = useState(CRYPTOS_PER_PAGE);
```
Comment (IoanSava) [inline cryk-ui/src/pages/Cryptocurrencies.tsx]: I don't think we need a state for that. `CRYPTOS_PER_PAGE` constant is enough.

File: cryk-ui/src/pages/Cryptocurrencies.tsx [hunk 2]
```diff
@@ -80,6 +92,13 @@ export default function Cryptocurrencies() {
                             <CryptoCard cryptocurrency={cryptocurrency} key={index} />
                         ))}
                     </div>
+                    <Pagination className="pagination"
+                        count={count}
+                        color="primary"
+                        size="large"
+                        page={currentPage}
+                        variant="outlined"
+                        onChange={(event, value) => handleChange(value)} />
```
Comment (IoanSava) [inline cryk-ui/src/pages/Cryptocurrencies.tsx]:
```javascript
onChange={(event, value) => setCurrentPage(value)}
```

File: cryk-ui/src/pages/Cryptocurrencies.tsx [hunk 3]
```diff
@@ -55,6 +61,12 @@ export default function Cryptocurrencies() {
         )
     }
 
+    const handleChange = (pageNumber: number) => {
+        setCurrentPage(pageNumber);
+    };
+
+    const count =  Math.ceil(data.cryptocurrenciesInfo.totalCount / cryptosPerPage);
```
Comment (IoanSava) [inline cryk-ui/src/pages/Cryptocurrencies.tsx]: I think we'll need a state for that variable and we'll update it within `useEffect`:
```javascript   
 useEffect(() => {
        if (data) {
            setCryptocurrencies(data.cryptocurrencies);
            setTotalNumberOfCryptocurrencies(Math.ceil(data.cryptocurrenciesInfo.totalCount / CRYPTOS_PER_PAGE));
        }
    }, [loading, data]);
```

File: cryk-ui/src/pages/Cryptocurrencies.tsx [hunk 4]
```diff
@@ -14,20 +14,27 @@ const GET_PAGINATED_CRYPTOCURRENCIES_QUERY = gql`
...
+        }
     }
 `;
 
+const CRYPTOS_PER_PAGE = 10;
+const CURRENT_PAGE = 1;
+
 export default function Cryptocurrencies() {
     const title = "Cryptocurrencies";
+    const [currentPage, setCurrentPage] = useState(CURRENT_PAGE);
```
Comment (IoanSava) [inline cryk-ui/src/pages/Cryptocurrencies.tsx]: We can remove `CURRENT_PAGE` constant and change it to:
```javascript
const [currentPage, setCurrentPage] = useState(1);
```

**Review by IoanSava — APPROVED** [review]

**Review by CipDan — APPROVED** [review]

**Pull request merged**